prov/shm/src/smr_progress.c

```diff
@@ -1062,18 +1149,65 @@ static void smr_progress_cmd(struct smr_ep *ep)
 	}
 }
 
+static void smr_progress_ipc_list(struct smr_ep *ep)
+{
+	struct smr_pend_entry *ipc_entry;
+	struct smr_domain *domain;
+	struct dlist_entry *tmp;
+	int ret;
+
+	domain = container_of(ep->util_ep.domain, struct smr_domain,
+			      util_domain);
+
+	pthread_spin_lock(&ep->region->lock);
+
+	/* after the synchronize all operations should be complete */
+	dlist_foreach_container_safe(&ep->ipc_cpy_pend_list,
+				     struct smr_pend_entry,
+				     ipc_entry, entry, tmp) {
+		enum fi_hmem_iface iface =
+			ipc_entry->cmd.msg.data.ipc_info.iface;
+		uint64_t device = ipc_entry->cmd.msg.data.ipc_info.device;
+
+		if (ofi_async_copy_query(iface, ipc_entry->async_event)
+				 != FI_SUCCESS)
```

Comment:
Nit - move operand (!=) to first line before new line. Or I think you should just be able to check
`if (!ofi_async_copy_query(iface, ipc_entry->async_event))`